### PR TITLE
Allow toolbox containers to be created with a custom hostname

### PR DESCRIPTION
**Pull request by nievesmontero**
Based on the following pull request https://github.com/containers/toolbox/pull/1007

### CODE REVIEW

**Comment by debarshiray** [comment]
> Based on the following pull request https://github.com/containers/toolbox/pull/1007

What's the difference from https://github.com/containers/toolbox/pull/1007 ?  Is this a rebase against `main`?

**Review by debarshiray — CHANGES_REQUESTED** [review]
The tests are failing:
```
not ok 25 create: Create a container with a custom hostname ('test.host-name.local') in 530ms
# (from function `assert_success' in file test/system/libs/bats-assert/src/assert.bash, line 114,
#  in test file test/system/101-create.bats, line 102)
#   `assert_success' failed
#
# -- command failed --
# status : 1
# output : Error: invalid hostname
# --
#
```

**Comment by debarshiray** [comment]
Also, there are several issues and pull requests discussing various approaches to improve the host name.  For example, did you see https://github.com/containers/toolbox/pull/1086 that seems to implement https://github.com/containers/toolbox/issues/969 ?

I am curious why you picked https://github.com/containers/toolbox/pull/1007

I also remember that @allanday had some ideas about the host name, but I can't find a reference right now.

File: src/cmd/create.go [hunk 1]
```diff
@@ -113,6 +121,10 @@ func init() {
 }
 
 func create(cmd *cobra.Command, args []string) error {
+	// This regex filters out strings which are not valid hostnames, according to RFC-1123.
+	// Source: https://stackoverflow.com/a/106223
+	var hostnameRegexp = regexp.MustCompile(`^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])$`)
```
Comment (debarshiray) [inline src/cmd/create.go:126]: Do we really need to have this regular expression directly in the code?  Can't we use some Go package for it?

**Comment by nievesmontero** [comment]
> > Based on the following pull request #1007
> 
> What's the difference from #1007 ? Is this a rebase against `main`?

Exactly, this new pull request is #1007 rebased against main. As #1007 was a bit old, there were conflicts I had to solve in order to be able to rebase it.

**Comment by nievesmontero** [comment]
> Also, there are several issues and pull requests discussing various approaches to improve the host name. For example, did you see #1086 that seems to implement #969 ?
> 
> I am curious why you picked #1007
> 
> I also remember that @allanday had some ideas about the host name, but I can't find a reference right now.

This pull request was picked randomly, trying to solve older pull requests.

**Comment by akdev1l** [comment]
@nievesmontero I would be partial towards picking the newest PR to merge instead of the oldest one

disclaimer: I'm the guy behind #1086 - totally willing to get this merged if you need some help

**Comment by nievesmontero** [comment]
@akdev1l thank you for pointing this out. I have tried rebasing https://github.com/containers/toolbox/pull/1086 and it seems to work correctly. I will reach you if I need any help, thanks.

**Comment by debarshiray** [comment]
> > > Based on the following pull request #1007
> > 
> > 
> > What's the difference from #1007 ? Is this a rebase against `main`?
> 
> Exactly, this new pull request is #1007 rebased against main. As #1007
> was a bit old, there were conflicts I had to solve in order to be able to
> rebase it.

I see.

Among other things, the tests are failing for this pull request, while they were passing for https://github.com/containers/toolbox/pull/1007

**Comment by debarshiray** [comment]
> > Also, there are several issues and pull requests discussing various approaches to improve the host name. For example, did you see #1086 that seems to implement #969 ?
> > I am curious why you picked #1007
> > I also remember that @allanday had some ideas about the host name, but I can't find a reference right now.
> 
> This pull request was picked randomly, trying to solve older pull requests.

I see.  This can't be solved *randomly*.  :)

We have two broad issues.  One is about improving the default hostname for the containers (see https://github.com/containers/toolbox/issues/969), and the other is about making it easier to visually distinguish between containers (see https://github.com/containers/toolbox/issues/98).

I suggest going through those two issues, and all the other issues and pull requests linked from them to get a good idea of the current situation.  That will give you a good idea of what needs to be done.

**Comment by debarshiray** [comment]
I am going to have to close this pull request because the comments from https://github.com/containers/toolbox/pull/1007 apply to this one too.

**Comment by buck-ross** [comment]
As the author of #1007, I agree with the decision to close both that PR and this one. #1086 seems to be a much simpler solution which addresses my initial issue without over-complicating things. Not to mention that the passage of time appears to have made integrating my original changes significantly more difficult.

**Pull request closed without merging**